src/modules/flow/http-server/http-server.c

```diff
@@ -0,0 +1,622 @@
+#define HTTP_HEADER_CONTENT_TYPE_JSON "application/json"
+
+struct http_data {
+    struct sol_flow_node *node;
+    union {
+        struct sol_irange i;
+        struct sol_drange d;
+        bool b;
+        char *s;
+    } value;
+
+    char *path;
+};
+
+static struct sol_http_server *server = NULL;
+static int init_count = 0;
+
+static int
+start_server(struct http_data *http, const char *path,
+    int (*handler)(void *data, struct sol_http_request *request))
+{
+    int r;
+
+    if (!server) {
+        server = sol_http_server_new(HTTP_SERVER_PORT);
+        SOL_NULL_CHECK(server, -1);
+    }
+
+    init_count++;
+
+    http->path = strdup(path);
+    SOL_NULL_CHECK_GOTO(http->path, err);
+
+    r = sol_http_server_register_handler(server, http->path, handler, http);
+    SOL_INT_CHECK_GOTO(r, < 0, err_handler);
+
+    return 0;
+
+err_handler:
+    free(http->path);
+err:
+    init_count--;
+    if (!init_count) {
+        sol_http_server_del(server);
+        server = NULL;
+    }
+    return -1;
+}
+
+static void
+stop_server(struct http_data *http)
+{
+    sol_http_server_unregister_handler(server, http->path);
+    free(http->path);
+    init_count--;
+    if (!init_count) {
+        sol_http_server_del(server);
+        server = NULL;
+    }
+}
+
+static int
+boolean_response_cb(void *data, struct sol_http_request *request)
+{
+    int r = 0;
+    uint16_t idx;
+    bool send_json = false;
+    enum sol_http_method method;
+    struct http_data *mdata = data;
+    char str[512] = { 0 };
+    struct sol_http_param_value *value;
+    struct sol_http_response response = {
+        .api_version = SOL_HTTP_RESPONSE_API_VERSION,
+        .content = SOL_BUFFER_INIT_EMPTY,
+        .param = SOL_HTTP_REQUEST_PARAM_INIT,
+        .response_code = SOL_HTTP_STATUS_OK
+    };
+
+    method = sol_http_request_get_method(request);
+    response.url = sol_http_request_get_url(request);
+
+    SOL_HTTP_PARAM_FOREACH_IDX (sol_http_request_get_params(request), value, idx) {
+        switch (value->type) {
+        case SOL_HTTP_PARAM_POST_FIELD:
+            if (streq(value->value.key_value.key, "value")) {
+                if (streq(value->value.key_value.value, "true"))
+                    mdata->value.b = true;
+                else if (streq(value->value.key_value.value, "false"))
+                    mdata->value.b = false;
+
+                r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+                SOL_INT_CHECK(r, < 0, r);
+            }
+            break;
+        case SOL_HTTP_PARAM_HEADER:
+            if (streq(value->value.key_value.key, HTTP_HEADER_ACCEPT)) {
+                if (strstr(value->value.key_value.value, HTTP_HEADER_CONTENT_TYPE_JSON))
+                    send_json = true;
+            }
+            break;
+        default:
+            break;
+        }
+    }
+
+    if (send_json) {
+        r = snprintf(str, sizeof(str), "{\n\t\"%s\": %s\n}", mdata->path,
```

Comment:
Nitpick, but there's no need to issue `\n\t` or other spaces in the JSON: just send `{"%s":%s}`.